google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java

```diff
@@ -141,93 +185,27 @@ public String buildSdkMessage(String path, boolean htmlEnabled) {
     return null;
   }
 
-  @VisibleForTesting
-  protected void showWarning(final String message) {
-    invokePanelValidationUpdate(new Runnable() {
-      @Override
-      public void run() {
-        warningMessage.setText(message);
-        warningMessage.setVisible(true);
-        warningIcon.setVisible(true);
-        cloudSdkDirectoryField.getTextField().setForeground(JBColor.red);
-      }
-    });
+  private void showWarning(String message) {
+    ApplicationManager.getApplication()
+        .invokeAndWait(
+            () -> {
+              warningMessage.setText(message);
+              warningMessage.setVisible(true);
+              warningIcon.setVisible(true);
+              cloudSdkDirectoryField.getTextField().setForeground(JBColor.red);
+            },
+            ModalityState.any());
   }
 
-  @VisibleForTesting
-  protected void hideWarning() {
-    invokePanelValidationUpdate(new Runnable() {
-      @Override
-      public void run() {
-        cloudSdkDirectoryField.getTextField().setForeground(JBColor.black);
-        warningIcon.setVisible(false);
-        warningMessage.setVisible(false);
-      }
-    });
-  }
-
-  private void invokePanelValidationUpdate(Runnable runnable) {
-    ApplicationManager.getApplication().invokeLater(runnable, ModalityState.any());
-  }
-
-  @VisibleForTesting
-  public TextFieldWithBrowseButton getCloudSdkDirectoryField() {
-    return cloudSdkDirectoryField;
-  }
-
-  @VisibleForTesting
-  JTextPane getWarningMessage() {
-    return warningMessage;
-  }
-
-  public boolean isModified() {
-    CloudSdkService sdkService = CloudSdkService.getInstance();
-
-    String cloudSdkDirectoryFieldValue =
-        getCloudSdkDirectoryText() != null ? getCloudSdkDirectoryText() : "";
-    String currentCloudSdkDirectory = sdkService.getSdkHomePath() != null
-        ? sdkService.getSdkHomePath().toString() : "";
-    return !cloudSdkDirectoryFieldValue.equals(currentCloudSdkDirectory);
-  }
-
-  public void apply() throws ConfigurationException {
-    CloudSdkService sdkService = CloudSdkService.getInstance();
-
-    if (sdkService.validateCloudSdk(getCloudSdkDirectoryText())
-        .contains(CloudSdkValidationResult.MALFORMED_PATH)) {
```

Comment:
also to document what we discussed offline - I think there's some inconsistency here with the validation: the original intent of the CloudSdkPanel was that it would never throw one of those ConfigurationExceptions. It would be up to consumers to decide if the validation would be blocking (and we are moving in the direction of having none of them block).

This can be saved for a separate PR since it wasn't introduced here (it was added in #1122).